src/tike/ptycho/solvers/rpie.py

```diff
@@ -356,14 +359,17 @@ def _update_wavefront(
     varying_probe,
     scan,
     psi,
+    exitwave_options,
     position_options,
     op=None,
 ):
     farplane = op.fwd(probe=varying_probe, scan=scan, psi=psi)
+
     intensity = cp.sum(
         cp.square(cp.abs(farplane)),
         axis=list(range(1, farplane.ndim - 2)),
     )
+
     cost = getattr(tike.operators, f'{op.propagation.model}_each_pattern')(
         data,
         intensity,
```

Comment:
Some of the convergence plots in the tests show non-convergence with more iterations, but I think this may be untrue. It may be because the object value is being computed over all pixels instead of just the measured pixels. i.e. the measured_pixels mask needs to be used here.

![convergence](https://github.com/AdvancedPhotonSource/tike/assets/9604511/4f17bf26-8d3f-44fe-85d3-4dbb9900fbf1)

Reply:
I think using intensity * exitwave_options.measured_pixels should fix this?

Reply:
see new commit

Reply:
No, it should be something like this:
```
    cost = getattr(tike.operators, f'{op.propagation.model}_each_pattern')(
        data[measured_pixels],
        intensity[measured_pixels],
```
The objective should only consider the measured pixels. If you do this:
```
    cost = getattr(tike.operators, f'{op.propagation.model}_each_pattern')(
        data,
        intensity * measured_pixels,
```
Then at the unmeasured pixels, you are comparing data[unmeasured_pixels] with 0, but data[unmeasured_pixels] could be nonzero!

Reply:
The "data" ND array should already have the unmeasured pixel regions zeroed out in the raw measurements, or at least it *should*.

What are you assuming the data with unmeasured pixels contains in the unmeasured pixels region?

So it should only be necessary to zero out those regions in the calculated intensities, i.e. the "intensity" variable.

Alternatively, when computing the Poisson or Gaussian cost function as a performance metric, we should maybe pass the measured pixels as a function argument here?

And do something like:

cost = sum_q  measured pixels * (  intensity - data * log( intensity ))

Reply:
> The "data" ND array should already have the unmeasured pixel regions zeroed out in the raw measurements.

As mentioned below, that shouldn't be a required pre-processing step that users need to do because it is extra pre-processing and it is ambiguous because zero is a valid photon count.

> What are you assuming the data with unmeasured pixels contains in the unmeasured pixels region?

It could contain any value; i.e. any value represented by the float and integer types including non-finite values, NaN, negative numbers, the largest or smallest integer, etc

> Alternatively, when computing the Poisson or Gaussian cost function as a performance metric, we should maybe pass the measured pixels as a function argument here?
>
> And do something like:
>
> cost = sum_q measured pixels * ( intensity - data * log( intensity ))

I don't see how this alternative addresses the problem that unmeasured pixels are being used to compute the objective function / performance metric.

Reply:
I think I'm doing what you wanted now?